### PR TITLE
Feat/decouple rendering

### PR DESCRIPTION
**Pull request by sncalvo**

### CODE REVIEW

File: GraficaEngine/GraficaEngine/Core/BaseGameObject.cpp [hunk 1]
```diff
@@ -71,6 +71,11 @@ namespace Engine
 		child->setParent(this);
 		_children.push_back(child);
 		child->start();
+
+		if (_scene)
```
Comment (franciscoaguirre) [inline GraficaEngine/GraficaEngine/Core/BaseGameObject.cpp:75]: We don't check the scene before in this function 🤔

File: GraficaEngine/GraficaEngine/Core/BaseGameObject.cpp [hunk 2]
```diff
@@ -97,6 +102,11 @@ namespace Engine
 				child),
 			_children.end());
 
+		if (_scene)
```
Comment (franciscoaguirre) [inline GraficaEngine/GraficaEngine/Core/BaseGameObject.cpp:105]: Is these needed?

File: GraficaEngine/GraficaEngine/Core/GameObject.cpp [hunk 3]
```diff
@@ -35,16 +43,14 @@ namespace Engine
 		return new GameObject(this);
 	}
 
-	void GameObject::draw() const
+	void GameObject::draw(Shader *shader) const
```
Comment (franciscoaguirre) [inline GraficaEngine/GraficaEngine/Core/GameObject.cpp:46]: Should we keep this around commented?

File: GraficaEngine/GraficaEngine/Core/Scene.h [hunk 4]
```diff
@@ -13,13 +13,17 @@
...
+#include "../Renderer/TextRenderer.h"
+
 namespace Engine
 {
 	class Scene
 	{
 	private:
 		Camera *_activeCamera;
-		std::map<std::string, Camera *> _cameras;
+		std::unordered_map<std::string, Camera *> _cameras;
```
Comment (franciscoaguirre) [inline GraficaEngine/GraficaEngine/Core/Scene.h:26]: I'm intrigued, why is it needed that this is an unordered map? The map is ordered by default?

**Review by franciscoaguirre — APPROVED** [review]

File: GraficaEngine/GraficaEngine/Core/Scene.h (hunk 4 above)
Comment (sncalvo) [reply GraficaEngine/GraficaEngine/Core/Scene.h:26]: The map is ordered by default. Causing the insertions to be done in order n

**Pull request merged**